### PR TITLE
sample: wifi: provisioning: adopt wifi_credentials lib

### PR DESCRIPTION
**Pull request by wentong-li**
Use wifi_crednentials lib to handle the reading/writing of wifi configuration. 
Also keep GATT service in common folder and move remaining part to user application.

### CODE REVIEW

**Comment by wentong-li** [comment]
@maxd-nordic Seems wifi_credentials lib does not have a doc page?

**Comment by maxd-nordic** [comment]
> @maxd-nordic Seems wifi_credentials lib does not have a doc page that I can link to?

It will have one very soon. It's included in this PR: https://github.com/nrfconnect/sdk-nrf/pull/9261

**Review by richabp — CHANGES_REQUESTED** [review]

File: samples/wifi/provisioning/common/wifi_prov_handler.c [hunk 1]
```diff
@@ -59,28 +76,32 @@ static void prov_get_status_handler(Request *req, Response *rsp)
 
 	rsp->has_device_status = true;
 
+	wifi_credentials_for_each_ssid(get_prov_info, &config);
+	if (config.header.ssid_len == 0) { /* No entry in wifi cred lib */
```
Comment (maxd-nordic) [inline samples/wifi/provisioning/common/wifi_prov_handler.c:80]: no need for this check, but you can keep it for peace-of-mind

**Comment by wentong-li** [comment]
Confirmed with @krish2718 that net_mgmt API call from system workqueue causes fatal error. So move the task that updates the advertisement data to a dedicated workqueue.

**Review by bama-nordic — APPROVED** [review]
LGTM. We should ensure fix for https://projecttools.nordicsemi.no/jira/browse/NRF7X-29 is tested (potentially from another PR).

**Comment by wentong-li** [comment]
Fix to NRF7X-29 applied.

File: samples/wifi/provisioning/src/main.c [hunk 2]
```diff
@@ -9,24 +9,263 @@
...
+		printk("Cannot update advertisement data, err = %d\n", rc);
+	}
+}
+
+static void connected(struct bt_conn *conn, uint8_t err)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	if (err) {
+		printk("BT Connection failed (err 0x%02x).\n", err);
```
Comment (krish2718) [inline samples/wifi/provisioning/src/main.c:103]: Any specific reason for switching to printk?

**Review by krish2718 — APPROVED** [review]
LGTM, it would be good to divide in to smaller commits for easier review and maintenance.

File: samples/wifi/provisioning/src/main.c (hunk 2 above)
Comment (wentong-li) [reply samples/wifi/provisioning/src/main.c:103]: This part is in user application, so I use printk here

File: samples/wifi/provisioning/implementation_details.rst [hunk 3]
```diff
@@ -190,9 +190,6 @@ Configuration management
...
 
 The configuration management component manages Wi-Fi configurations.
-It ensures that the data in RAM and flash are in sync, and exposes APIs to add, read, or remove configurations in a multi-thread environment.
+It uses wifi_credentials library to handle the configurations in flash. It also has one slot in RAM to save the configurations.
 
-The Wi-Fi Provisioning Service offloads all configuration management operations to the configuration management component, and only uses the APIs the component provides to change the configuration.
-
-This component saves data in non-volatile memory, which is the internal flash for nRF5x devices.
-You can save the configuration in both RAM and flash or only in RAM during provisioning.
+You can save the configuration in flash or RAM during provisioning.
```
[the remark this answers is not in the document]
Comment (wentong-li) [reply samples/wifi/provisioning/implementation_details.rst:196]: A newline is already there. And it does go to a separate paragraph in doc preview.

**Review by krga2022 — APPROVED** [review]

**Review by richabp — APPROVED** [review]

**Pull request merged**